parser/src/string.rs

```diff
@@ -0,0 +1,189 @@
+use crate::ast::{Constant, Expr, ExprKind};
+use crate::error::{LexicalError, LexicalErrorType};
+
+pub fn parse_implicit_concatenation(values: Vec<Expr>) -> Result<Expr, LexicalError> {
```

Comment:
The new signature will look like:
```suggestion
pub fn parse_implicit_concatenation(values: Vec<(Location, (String, StringKind))>) -> Result<Expr, LexicalError> {
```